src/sentry/snuba/metrics_enhanced_performance.py

```diff
@@ -70,19 +72,22 @@ def query(
                 limit=limit,
                 offset=offset,
             )
-            # Getting the 0th result for now, will need to consolidate multiple query results later
-            results = metrics_query.run_query(referrer + ".metrics-enhanced")
-            results = discover.transform_results(
-                results, metrics_query.function_alias_map, {}, None
-            )
-            results = resolve_tags(results, metrics_query)
-            results["meta"]["isMetricsData"] = True
-            return results
+            with sentry_sdk.start_span(op="mep", description="query.transform_results"):
+                # Getting the 0th result for now, will need to consolidate multiple query results later
+                results = metrics_query.run_query(referrer + ".metrics-enhanced")
+                results = discover.transform_results(
+                    results, metrics_query.function_alias_map, {}, None
+                )
+                results = resolve_tags(results, metrics_query)
+                results["meta"]["isMetricsData"] = True
+                return results
         # raise Invalid Queries since the same thing will happen with discover
         except InvalidSearchQuery as error:
             raise error
         # any remaining errors mean we should try again with discover
-        except IncompatibleMetricsQuery:
+        except IncompatibleMetricsQuery as error:
+            sentry_sdk.set_tag("performance.use_metrics", False)
```

Comment:
Maybe we have a second tag here whether metrics was used or not and we keep it separate from the param

Reply:
Hmm my thinking here was the metrics_incompatible tag lets us know that it was attempted but not used

Reply:
:+1: added `performance.dataset` so there's an explicit tag about which we use